src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,77 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkBase;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj2.command.*;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.lib.Neo;
+
+import java.util.function.DoubleSupplier;
+
+import static frc.robot.Constants.IntakeConstants.*;
+
+public class Intake extends SubsystemBase {
+    public int targetPos = INTAKE_ANGLE.IDLE.angle;
+    private final PIDController anglePIDcontroller = new PIDController(PIDGains.kp, PIDGains.ki, PIDGains.kd);
+    private final ArmFeedforward angleFFcontroller = new ArmFeedforward(FFangleGains.ks, FFangleGains.kg, FFangleGains.kv);
+    private final Neo intakeMotor = new Neo(0);
+    private final Neo angleMotor = new Neo(0);
+    private final DutyCycleEncoder intakeEncoder = new DutyCycleEncoder(0);
+    private final DigitalInput beamBreak = new DigitalInput(0);
+    private final Trigger beamBreakTrigger = new Trigger(()-> beamBreak.get());
+    public static final Intake intake = new Intake(); //that made you laugh right?
+    public Intake(){
+        intakeMotor.setIdleMode(CANSparkBase.IdleMode.kBrake);
+        intakeMotor.setInverted(false);
+        angleMotor.setInverted(false);
+        angleMotor.setIdleMode(CANSparkBase.IdleMode.kBrake);
+        intakeMotor.setConversionFactors(0,0);
+        intakeEncoder.setDistancePerRotation(360);
+        intakeEncoder.setPositionOffset(0);
+    }
+    public Command rollerSpeedCommand(DoubleSupplier speed){
+        return new RunCommand(()-> intakeMotor.set(speed.getAsDouble())).until(beamBreakTrigger);
+    }
+    public Command stopRollersCommand(){
+        return new RunCommand(()-> intakeMotor.set(0));
+    }
+    public double getAngle(){
```

Comment:
We get the angle from the absolute encoder not the neo one...

Reply:
this is a duty cycle encoder if you look in the top of the class, but I will name it more clearly